tests/ui/feature-gates/feature-gate-pin_ergonomics.rs

```diff
@@ -4,12 +4,20 @@ use std::pin::Pin;
 
 struct Foo;
 
+impl Foo {
+    fn foo(self: Pin<&mut Self>) {
+    }
+}
+
 fn foo(_: Pin<&mut Foo>) {
 }
 
 fn bar(mut x: Pin<&mut Foo>) {
     foo(x);
     foo(x); //~ ERROR use of moved value: `x`
+
+    x.foo(); //~ ERROR use of moved value: `x`
+    x.foo(); //~ ERROR use of moved value: `x`
 }
```

Comment:
Split this into a separate test, so that the first `x.foo()` call doesn't fail, but the second does.

Reply:
I'll break it out into a separate function in the same file so it's still one test for the feature gate overall.